core/Http.php

```diff
@@ -195,13 +196,20 @@ public static function sendHttpRequestBy(
             throw new Exception('Too many redirects (' . $followDepth . ')');
         }
 
+
         $aUrl = preg_replace('/[\x00-\x1F\x7F]/', '', trim($aUrl));
         $parsedUrl = @parse_url($aUrl);
 
         if (empty($parsedUrl['scheme'])) {
             throw new Exception('Missing scheme in given url');
         }
 
+        //secure matomo request in https
+        if (GeneralConfig::getConfigValue('enable_ssl_request') && preg_match('/^[a-zA-Z]+.matomo.org/',
+            $parsedUrl['host']) && $parsedUrl['scheme'] == 'http') {
+            $parsedUrl['scheme'] = 'https';
+        }
```

Comment:
Shouldn't instead of hardcoding the protocol change (and therefore making what the rest of the code does more confusing) we instead change the actual places where HTTP fallbacks and HTTP-by-default URLs are still used?

So e.g. changing api_service_url, remove things like
https://github.com/matomo-org/matomo/blob/c11469e4199405628e97650914f0cdbdea94dac7/plugins/CoreUpdater/Diagnostic/HttpsUpdateCheck.php#L38-L40
and all usages of 
https://github.com/matomo-org/matomo/blob/009353e3db72ca214eaf32a9596a52eed594eb5f/plugins/CoreUpdater/Controller.php#L455
, etc.

So I think we should remove the ability for people to easily shoot themselves into the foot instead of changing the behaviour in confusing ways (the insecure HTTP fallback that is still offered also failing with a SSL error) and adding complexity instead of using this as a chance to get rid of a lot of insecure fallback-code.